Remove ItemFactory#enchantWithLevels range check for vanilla parity

Esophose opened the pull request:
Resolves #12208 by removing the level range check on ItemFactory#enchantWithLevels. The old ItemFactory#enchantItem did not have a level range, and the vanilla method does not have a level range either. There isn't really any reason to have this. If desired, it could be capped with a lower bound of 1, but it doesn't seem necessary.

Vanilla has at least one loot table that uses a level above 30, see the end city treasure table.

Esophose committed:
Remove ItemFactory#enchantWithLevels range check for vanilla parity #12208

Warriorrrr reviewed (commented):
I think that it's worth noting that you can only use levels 1-30 in a normal enchanting table, and that at higher levels (>= 100) you will not be able to get any enchants in vanilla.

Esophose committed:
Add comments detailing enchantWithLevels `levels` parameter

Owen1212055 committed:
Javadoc tweaks

Removed this because technically you can register your own enchantments with large enough values to be picked

Owen1212055 committed:
Update ItemStack.java
